### PR TITLE
Platform/angela/14331/add loinc to azure events

### LINKED ISSUE #14331: Update FHIRRouter Azure Events to Store LOINC Code Info
### User Story

As a ReportStream developer troubleshooting issues,
I want to see the LOINC code alongside the condition ReportStream mapped it to,
so I can evaluate its accuracy when needed.

### Description/Use Case

Update `AzureEventUtils.getObservations` to include the `code` and `display` fields in the second to last box pictured below

image

### Acceptance Criteria

- [ ] `ReportAcceptedEvent` and `ReportRoutedEvent` both include test performed LOINC code and display alongside the mapped condition info

### PR DESCRIPTION
**Pull request by adegolier**
This PR ...

Adds the original code and display that mapped conditions came from to ObservationSummary in the logs

Test Steps:
1. Run AzureEventUtilsTest.

Alternative test:
1. Set up local environment to point to appinsights in test.
2. Send a message that will be routed via postman or your local submission route of choice.
3. View the events in test appinsights in Azure.

## Changes
- Added code and display to ObservationSummary
- Added code and display to be put into ObservationSummary into AzureEventUtils
- Updated tests  
- Created new test file that is a modified copy of valid.fhir

### Testing
- [X] Tested locally?
- [X] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [X] Added/Updated tests?

## Linked Issues
- Fixes #14331 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

### CODE REVIEW

File: prime-router/src/main/kotlin/azure/observability/event/TestSummary.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package gov.cdc.prime.router.azure.observability.event
+
+import gov.cdc.prime.router.fhirengine.utils.conditionCodeExtensionURL
+import org.hl7.fhir.r4.model.Coding
+
+data class TestSummary(
+    val conditions: List<CodeSummary> = emptyList(),
+    val testPerformedSystem: String = "Unknown",
```
Comment (david-navapbc) [inline prime-router/src/main/kotlin/azure/observability/event/TestSummary.kt:8]: nit - this value can be a const to keep the code DRY

File: prime-router/src/test/kotlin/fhirengine/engine/FhirDestinationFilterTests.kt [hunk 2]
```diff
@@ -323,17 +324,55 @@ class FhirDestinationFilterTests {
...
+                                "94558-4",
+                                "SARS-CoV-2 (COVID-19) Ag [Presence] in Respiratory specimen by Rapid immunoassay"
+                            )
+                        )
+                    ),
+                    ObservationSummary(
+                        listOf(
+                            TestSummary(
+                                testPerformedCode = "95418-0",
+                                testPerformedSystem = "http://loinc.org"
```
Comment (david-navapbc) [inline prime-router/src/test/kotlin/fhirengine/engine/FhirDestinationFilterTests.kt:346]: same nit here

**Review by david-navapbc — APPROVED** [review]
code lgtm. developer hopped on a call with me and gave me a live demo of the changes working as per ACs. Approved and excellent work!

File: prime-router/src/test/kotlin/fhirengine/engine/FhirDestinationFilterTests.kt (hunk 2 above)
Comment (adegolier) [reply prime-router/src/test/kotlin/fhirengine/engine/FhirDestinationFilterTests.kt]: I think I got them all 🤞

File: prime-router/src/test/kotlin/azure/observability/event/AzureEventUtilsTest.kt [hunk 3]
```diff
@@ -18,15 +19,53 @@ class AzureEventUtilsTest {
 
         val expected = listOf(
             ObservationSummary(
-                ConditionSummary(
-                    "840539006",
-                    "Disease caused by severe acute respiratory syndrome coronavirus 2 (disorder)"
+                listOf(
+                    TestSummary(
+                        listOf(
+                            CodeSummary(
```
Comment (arnejduranovic) [inline prime-router/src/test/kotlin/azure/observability/event/AzureEventUtilsTest.kt:25]: Can we add a case where there are multiple test performed codes in a single observation?

**Review by arnejduranovic — CHANGES_REQUESTED** [review]

File: prime-router/src/test/kotlin/azure/observability/event/AzureEventUtilsTest.kt (hunk 3 above)
Comment (adegolier) [reply prime-router/src/test/kotlin/azure/observability/event/AzureEventUtilsTest.kt:27]: Done. Added some other cases, too.

**Review by arnejduranovic — APPROVED** [review]

**Pull request merged**